flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridge.java

```diff
@@ -348,14 +348,17 @@ public void notifyNotEnoughResourcesAvailable(
             Collection<ResourceRequirement> acquiredResources) {
         assertRunningInMainThread();
 
-        failPendingRequests();
+        failPendingRequests(acquiredResources);
     }
 
-    private void failPendingRequests() {
+    private void failPendingRequests(Collection<ResourceRequirement> acquiredResources) {
         if (!pendingRequests.isEmpty()) {
             final NoResourceAvailableException cause =
                     new NoResourceAvailableException(
-                            "Could not acquire the minimum required resources.");
+                            "Could not acquire the minimum required resources. Acquired: "
+                                    + acquiredResources
+                                    + ". Current slot pool status: "
+                                    + getSlotServiceStatus());
```

Comment:
Printing `acquiredResources` can actually produce quite some text. Hence, I am wondering whether we should really include this information in every `NoResourceAvailableException` since it will be logged for every request later on. Maybe it could be better to log it once in this method and then fail the requests w/o this information. 

```
log.info("Could not acquire the minimum required resources. Failing all pending requests. Acquired {}. Current slot pool status: {}.", acquiredResources, getSlotServiceStatus());
```

What do you think @rkhachatryan?

Reply:
Opened #16457 after an offline discussion.